cmd/dtmate/cmd/diff.go

```diff
@@ -66,6 +68,17 @@ func getInput() (string, string) {
 	return line, end
 }
 
+// convert duration from one group of units to another
+func convDuration(source, target string, brief bool) string {
+	conv := DateTimeMate.NewConv(DateTimeMate.ConvWithSource(source), DateTimeMate.ConvWithTarget(target), DateTimeMate.ConvWithBrief(brief))
+	result, err := conv.ConvertDuration()
+	if err != nil {
+		fmt.Fprintln(os.Stderr, err)
+		os.Exit(1)
+	}
+	return result
+}
```

Comment:
There is a problem with the signature

You should never do an exit in a function, only in main.go

Also printing the error here is not a good idea

So here I would return an error via the signature.

I see the method is called from outputDiff
Its signature should be altered

You should have no os.Exit or fmt.Println(err) except in main.go

Otherwise, you cannot test it.

Also, you are using cobra. So you could simply use RunE instead of Run to be able to return an error, cobra will exit if you return an error.

But this is a design problem way wider spread than this PR.

It should be addressed in a dedicated issue and PR that will be about refactoring ypur code

Reply:
Said otherwise, you can merge this PR, almost like this, but then you will have to refactor

Reply:
I like the idea of merging this and then opening a dedicated PR to fix the `os.Exit` problem that you mentioned. Thanks for pointing out `RunE` -- I did not know about that.

Reply:
You are welcome. I saw a lot of tool using cargo dealing with door like you do now because they dont know they can do it